From: Shiji Yang <yangshiji66@outlook.com>
Date: Wed, 31 May 2023 00:15:23 +0000
Subject: [PATCH] ath79: ignore the abused interrupt-map on pcie node

ath79 PCIe interrupt controller has stopped working correctly. This
is because the DT exposing a non-sensical interrupt-map property,
and their drivers relying on the kernel ignoring this property[1].

This patch fix the pcie init error:
ath9k 0000:00:00.0: of_irq_parse_pci: failed with rc=-14

Notice:
This is just a workaround, not a fix. PCIe driver and related dts
node need to be rewritten.

[1] https://lore.kernel.org/all/20211201114102.13446-1-maz@kernel.org/

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
---
 drivers/of/irq.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/of/irq.c
+++ b/drivers/of/irq.c
@@ -94,6 +94,8 @@ EXPORT_SYMBOL_GPL(of_irq_find_parent);
  * drawing board.
  */
 static const char * const of_irq_imap_abusers[] = {
+	"qca,ar7100-pci",
+	"qcom,ar7240-pci",
 	"CBEA,platform-spider-pic",
 	"sti,platform-spider-pic",
 	"realtek,rtl-intc",
